tests/e2etest/main.rs

```diff
@@ -613,12 +620,15 @@ async fn http_post_to_intf(
         .body(value_json.clone())
         .send()
         .await
-        .map_err(|e| format!("HTTP POST failure: {e}"))?
-        .text()
-        .await
-        .map_err(|e| format!("Failure in parsing the HTTP POST result: {e}"))?;
-    if response != value_json {
-        println!("Server response: {response}");
+        .map_err(|e| format!("HTTP POST failure: {e}"))?;
+    if response.status() != 200 {
```

Comment:
Can we replace 200 with the Reqwest status code enum?
```suggestion
    if response.status() != StatusCode::Ok{
```